### DISCUSSION TITLE
Next 13: Root Layout component Question

### DISCUSSION THREAD

**Discussion by snowytime** [vercel/next.js#42468] [Help]
Hey Nextjs folks,

With the new next13 features, if I set the root `layout.tsx` component to `"use client";` does that mean that all the pages bellow it will also be client side only?

**Comment by icyJoseph** [comment]
No, Server Components in that sub-tree create a sort of network boundary, so that while rendering on the client, when a SC is hit, a request for it's representation is sent to the server, loading and suspense boundaries can be used to show UI while this request resolves. **There's oversimplification and perhaps inaccuracies on that statement**, so take it with a grain of salt, I am still refining my mental model around "use client" and Server Components. It might be better to go off this statement:

> With this pattern, React will know it needs to render <ServerComponent /> on the server before sending the result (which doesn't contain any server-only code) to the client. From the Client Component's perspective, its child will be already rendered.
> 
> This pattern is already applied in [layouts and pages](https://beta.nextjs.org/docs/routing/pages-and-layouts) with the children prop so you don't have to create an additional wrapper component.

I think this is more of a performance consideration. If the layout is using "use client" then the layout will have some additional JavaScript, which you could totally skip if you instead refactor your app so that the "leaf" components in your application are clients. See these examples:

-  https://beta.nextjs.org/docs/rendering/server-and-client-components#importing-server-components-into-client-components - server components, with some restrictions, can be children of client components. 
- https://beta.nextjs.org/docs/rendering/server-and-client-components#moving-client-components-to-the-leaves 

When it comes to providers and such it is perhaps best to define a wrapper "use client" component which is then imported into an otherwise Server Component layout. See this example, https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components - Provider is use-client, and it is used to wrap layout children.
